services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityCredentialsProviderFactory.java

```diff
@@ -50,30 +50,39 @@
 public final class StsWebIdentityCredentialsProviderFactory implements WebIdentityTokenCredentialsProviderFactory {
 
     public AwsCredentialsProvider create(WebIdentityTokenCredentialProperties credentialProperties) {
-        return new StsWebIdentityCredentialsProvider(credentialProperties);
+        return new StsWebIdentityCredentialsProvider(credentialProperties, null);
+    }
+
+    public AwsCredentialsProvider create(WebIdentityTokenCredentialProperties credentialProperties,
+                                         StsClient stsClient) {
+        return new StsWebIdentityCredentialsProvider(credentialProperties, stsClient);
```

Comment:
Do we need to add this? Create methods are usually to make things more convenient for our customers, but this is a protected API that they won't use, so I'd prefer just to rely on the builder.